scraper/google.py

```diff
@@ -0,0 +1,218 @@
+        location_dict = {
+            "city": location_list[0],
+            "state": location_list[1],
+            "country": location_list[2],
+        }
+        job_locations.append(location_dict)
+    return job_locations
+
+
+def get_google_job_qualifications(job_post: BeautifulSoup):
+    """
+    Gets the job qualifications from a Google job posting.
+
+    Parameters
+    ----------
+    job_post : BeautifulSoup
+        A soupified html representation for a single Google job posting
+
+    Returns
+    -------
+    job_qualifications : list[str]
+        A list of qualifications for the job posting
+    """
+
+    job_qualifications = []
+    qualifications_div = job_post.find("div", {"itemprop": "qualifications"})
+    list_containers = qualifications_div.find_all("ul")
+    for ul in list_containers:
+        list_items = ul.find_all("li")
+        for item in list_items:
+            job_qualifications.append(item.text)
+    return job_qualifications
+
+
+def get_google_job_description(job_post: BeautifulSoup):
+    """
+    Gets the job description from a Google job posting.
+
+    Parameters
+    ----------
+    job_post : BeautifulSoup
+        A soupified html representation for a single Google job posting
+
+    Returns
+    -------
+    job_description : str
+        A string containing the entire job description labeled in the job
+        posting under 'About the job'
+    """
+
+    description_div = job_post.find("div", {"itemprop": "description"})
+    description_paragraphs = [p.text for p in description_div.find_all("p")]
+    string_separator = "\n"
+    job_description = string_separator.join(description_paragraphs)
+    return job_description
+
+
+def get_google_job_responsibilities(job_post: BeautifulSoup):
+    """
+    Gets the job responsibilities from a Google job posting.
+
+    Parameters
+    ----------
+    job_post : BeautifulSoup
+        A soupified html representation for a single Google job posting
+
+    Returns
+    -------
+    job_responsibilities : list[str]
+        A list of responsibilities given for the job post
+    """
+
+    # locate the div containing responsibilities and get the underlying ul element
+    responsibilities_list = job_post.find("div", {"itemprop": "responsibilities"}).find(
+        "ul"
+    )
+    job_responsibilities = [li.text for li in responsibilities_list.find_all("li")]
+    return job_responsibilities
+
+
+def parse_google_job_description(driver: Firefox, href: str):
+    """
+    Parse through a google job description, returning information from the
+    job description
+
+    Parameters
+    ----------
+    driver : (Firefox)WebDriver
+        A selenium web driver object
+    href : str
+        A reference link to a Google job posting
+
+    Returns
+    -------
+    job_info : dict{}
+        A dictionary containg all of the job information parsed from a
+        Google job posting
+    """
+
+    base = "https://careers.google.com"
```

Comment:
Can we push this base variable up to a constant or to a global variable? We use this domain url further up too I think

Reply:
sure thing